pkg/controller/pxc/version.go

```diff
@@ -98,26 +98,35 @@ func (r *ReconcilePerconaXtraDBCluster) ensurePXCVersion(cr *api.PerconaXtraDBCl
 		return errors.New("cluster is not ready")
 	}
 
-	new, err := vs.GetExactVersion(cr.Spec.UpgradeOptions.Apply, cr.Status.PXC.Version)
+	newVersion, err := vs.GetExactVersion(cr.Spec.UpgradeOptions.Apply, cr.Status.PXC.Version, currVersionMeta{
+		Platform:      string(r.serverVersion.Platform),
```

Comment:
it is needed to make sure that `cr.Platform` has priority over `r.serverVersion.Platform`